src/EnergyPlus/RoomAirModelManager.cc

```diff
@@ -2693,225 +2714,204 @@ namespace RoomAir {
-
-        MaxNums = max(MaxNums, NumNumbers);
-        MaxAlphas = max(MaxAlphas, NumAlphas);
-
-        if (MaxNums > NumNumbers) {
-            Numbers.allocate(MaxNums);
-            Numbers = 0.0;
-        } else if (!allocated(Numbers)) {
-            Numbers.allocate(MaxNums);
-        }
-
-        if (MaxAlphas > NumAlphas) {
-            Alphas.allocate(MaxAlphas);
-            Alphas = "";
-        } else if (!allocated(Alphas)) {
-            Alphas.allocate(NumAlphas);
-        }
-
-        for (int I = 1; I <= TotNumEquip; ++I) {
-            state.dataInputProcessing->inputProcessor->getObjectItem(state, equipTypeName, I, Alphas, NumAlphas, Numbers, NumNumbers, Status);
-            if (Util::SameString(Alphas(1), EquipName)) {
-                EquipFind = true;
-                break;
-            }
-        }
-
         switch (zoneEquipType) {
         case DataZoneEquipment::ZoneEquipType::VariableRefrigerantFlowTerminal: { // ZoneHVAC:TerminalUnit : VariableRefrigerantFlow
-            SupplyNodeName = Alphas(4);
-            ReturnNodeName = ""; // Zone return node
+            SupplyNodeNum = HVACVariableRefrigerantFlow::GetVRFTUZoneInletAirNode(state, EquipIndex);
         } break;
         case DataZoneEquipment::ZoneEquipType::EnergyRecoveryVentilator: { // ZoneHVAC : EnergyRecoveryVentilator
-            int nodeNum = GetFanOutletNode(state, "Fan:OnOff", Alphas(4), errorfound);
-            if (errorfound) {
-            }
-            SupplyNodeName = state.dataLoopNodes->NodeID(nodeNum); // ?????
-            ReturnNodeName = "";                                   // Zone exhaust node
+            SupplyNodeNum = GetFanOutletNode(state, "Fan:OnOff", EquipName, errorfound);
         } break;
         case DataZoneEquipment::ZoneEquipType::FourPipeFanCoil: { // ZoneHVAC : FourPipeFanCoil
-            SupplyNodeName = Alphas(6);
-            ReturnNodeName = Alphas(5);
+            SupplyNodeNum = state.dataFanCoilUnits->FanCoil(EquipIndex).AirOutNode;
+            ReturnNodeNum = state.dataFanCoilUnits->FanCoil(EquipIndex).AirInNode;
         } break;
         case DataZoneEquipment::ZoneEquipType::OutdoorAirUnit: { // ZoneHVAC : OutdoorAirUnit
-            SupplyNodeName = Alphas(13);
-            ReturnNodeName = Alphas(14);
+            SupplyNodeNum = state.dataOutdoorAirUnit->OutAirUnit(EquipIndex).AirOutletNode;
+            ReturnNodeNum = state.dataOutdoorAirUnit->OutAirUnit(EquipIndex).AirInletNode;
         } break;
         case DataZoneEquipment::ZoneEquipType::PackagedTerminalAirConditioner: { // ZoneHVAC : PackagedTerminalAirConditioner
-            SupplyNodeName = Alphas(4);
-            ReturnNodeName = Alphas(3);
+            SupplyNodeNum = state.dataUnitarySystems->unitarySys[EquipIndex - 1].getAirOutNode(state, EquipName, 0, errorfound);
+            ReturnNodeNum = state.dataUnitarySystems->unitarySys[EquipIndex - 1].getAirInNode(state, EquipName, 0, errorfound);
```

Comment:
Get inlet and outlet nodes from existing functions

Reply:
The problem here is that EquipIndex is the index to the type of PTUnit equipment (i.e., PTAC, PTHP, or PTWSHP) which is not the same as the index to the unitarySys struct.

                        // zone equipment require a 1-n index for access to zone availability managers
                        switch (getPTUnitType) {
                        case 1: // Excuse me?
                            ++numPTAC;
                            thisSys.m_EquipCompNum = numPTAC;
                            break;
                        case 2: // Baking powder?
                            ++numPTHP;
                            thisSys.m_EquipCompNum = numPTHP;
                            break;
                        case 3:
                            ++numPTWSHP;
                            thisSys.m_EquipCompNum = numPTWSHP;
                            break;
                        default:
                            assert(true);
                        }
                        int thisSysNum = state.dataUnitarySystems->numUnitarySystems - 1;
                        state.dataUnitarySystems->unitarySys[thisSysNum] = thisSys;

Reply:
@rraustad I am going to check it.

Reply:
@rraustad What I need is to find UnitarySys index from state.dataZoneEquip->ZoneEquipList

                    int EquipIndex = 0;
                    for (int thisZoneEquipNum = 1; thisZoneEquipNum <= state.dataZoneEquip->ZoneEquipList(iZone).NumOfEquipTypes;
                         ++thisZoneEquipNum) {
                        if (Util::SameString(state.dataZoneEquip->ZoneEquipList(iZone).EquipName(thisZoneEquipNum), roomAFNNodeHVAC.Name)) {
                            EquipIndex = state.dataZoneEquip->ZoneEquipList(iZone).EquipIndex(thisZoneEquipNum);
                        }
                    }
Then I look at UnitarySys inlet and outlet nodes. thisSys.m_EquipCompNum is irrelevant for this Check.

Reply:
thisSys.m_EquipCompNum is the value that gets passed back from `void UnitarySys::simulate` to the calling function. It is VERY relevant.

        CompIndex = this->m_EquipCompNum;